src/main/java/frc/team2412/robot/Subsystems/IndexerSubsystem.java

```diff
@@ -0,0 +1,304 @@
+	}
+	public void allShiftUp() {
+		rearClutch.set(DoubleSolenoid.Value.kForward);
+		frontClutch.set(DoubleSolenoid.Value.kForward);
+	}
+	public void allShiftDown() {
+		rearClutch.set(DoubleSolenoid.Value.kReverse);
+		frontClutch.set(DoubleSolenoid.Value.kReverse);
+	}
+	// Motors
+	public void stopAll() {
+		indexMotors.set(0.0);
+	}
+
+	public void stopSides() {
+		indexSideMotors.set(0.0);
+	}
+
+	// Sensors
+	public boolean allSensorsOn() {
+		if (back.get() && backMid.get() && mid.get() && frontMid.get() && front.get()) {
+			return true;
+		}
+
+		return false;
+		// check if all sensors are on
+	}
+
+	public boolean allSensorsOff() {
+		if (!back.get() && !backMid.get() && !mid.get() && !frontMid.get() && !front.get()) {
+			return true;
+		}
+
+		return false;
+		// check if all sensors are on
+	}
+
+	/*
+	 * if(allSensorsOff()){intake12();
+	 * 
+	 * boolean b;if(backMid.get()) { b = true; } else { b = false; }
+	 * switchBalls2(b); intake3(); if (backMid.get()) { b = true; } else { b =
+	 * false; } switchBalls34(b); intake4(); if (back.get()) { b = true; } else { b
+	 * = false; } switchBalls34(b); intake5(); }
+	 * 
+	 * shoot();
+	 * 
+	 * }
+	 */
+	public void shoot() {
+		allShiftDown();
+		indexMidMotor.set(1);
+		if (frontMid.get() && front.get()) {
+			indexFrontMotor.set(0);
+		} else if (backMid.get() && back.get()) {
+			indexBackMotor.set(0);
+		}
+		
+		stopAll();
+	}
+
+//Switch ball positions
+	public void switchBalls2(boolean currentDirection) {
+		if (currentDirection != (DriveBaseSubsystem.getCurrentYSpeed >= 0)) {
+			if (currentDirection) {
+				indexFrontMotor.set(1);
+				indexBackMotor.set(1);
+				if (!backMid.get()) {
+					stopAll();
+					ballUnbalancedSide = !ballUnbalancedSide;
+				}
+			} else {
+				indexFrontMotor.set(-1);
+				indexBackMotor.set(-1);
+				if (!frontMid.get()) {
+					stopAll();
+					ballUnbalancedSide = !ballUnbalancedSide;
+				}
+			}
+		}
+	}
+
+	public void switchBalls34(boolean currentDirection) {
+		if (currentDirection != (DriveBaseSubsystem.getCurrentYSpeed >= 0)) {
+			if (currentDirection) {
+				indexFrontMotor.set(1);
+				indexBackMotor.set(1);
+				if (!back.get()) {
+					stopAll();
+					ballUnbalancedSide = !ballUnbalancedSide;
+				}
+			} else {
+				indexFrontMotor.set(-1);
+				indexBackMotor.set(-1);
+				if (!front.get()) {
+					stopAll();
+					ballUnbalancedSide = !ballUnbalancedSide;
+				}
+			}
+		}
+	}
+
+//first two balls
+	public void intake1() {
+
+		indexFrontMotor.set(1);
+		indexBackMotor.set(-1);
+		if (!mid.get()) {
+			stopAll();
+			numBalls++;
+		}
+	}
+
+	public void intake2() {
+		indexFrontMotor.set(1);
+		indexBackMotor.set(-1);
+		if (!frontMid.get()) {
+			stopAll();
+			ballUnbalancedSide = false;
+			numBalls++;
+		} else if (!backMid.get()) {
+			stopAll();
+			ballUnbalancedSide = true;
+			numBalls++;
+		}
+	}
+
+//Third Ball 
+	private int run = 0;
+	private boolean intakeDirection;

```

Comment:
`true` and `false` aren't directions.  If you really want to use `boolean` instead of defining an enum in four lines of code, name this `goingForward` or `intakeFromBack`.